FScriptTE.pas

```diff
@@ -0,0 +1,825 @@
+        if opcodelist[i].arg[v] = T_REG then
+          argstring := argstring + 'T_REG'
+        else if opcodelist[i].arg[v] = T_RREG then
+          argstring := argstring + 'T_RREG'
+        else if opcodelist[i].arg[v] = T_BREG then
+          argstring := argstring + 'T_BREG'
+        else if opcodelist[i].arg[v] = T_DREG then
+          argstring := argstring + 'T_DREG'
+        else if opcodelist[i].arg[v] = T_SWITCH2B then
+          argstring := argstring + 'T_SWITCH2B'
+        else if opcodelist[i].arg[v] = T_SWITCH then
+          argstring := argstring + 'T_SWITCH'
+        else if opcodelist[i].arg[v] = T_SWITCHZ then
+          argstring := argstring + 'T_SWITCHZ'
+        else if opcodelist[i].arg[v] = T_BYTE then
+          argstring := argstring + 'T_BYTE'
+        else if opcodelist[i].arg[v] = T_WORD then
+          argstring := argstring + 'T_WORD'
+        else if opcodelist[i].arg[v] = T_DWORD then
+          argstring := argstring + 'T_DWORD'
+        else if opcodelist[i].arg[v] = T_DATA then
+          argstring := argstring + 'T_DATA'
+        else if opcodelist[i].arg[v] = T_STRDATA then
+          argstring := argstring + 'T_STRDATA'
+        else if opcodelist[i].arg[v] = T_PFLAG then
+          argstring := argstring + 'T_PFLAG'
+        else if opcodelist[i].arg[v] = T_FUNC then
+          argstring := argstring + 'T_FUNC'
+        else if opcodelist[i].arg[v] = T_FUNC2 then
+          argstring := argstring + 'T_FUNC2'
+        else if opcodelist[i].arg[v] = T_FLOAT then
+          argstring := argstring + 'T_FLOAT'
+        else if opcodelist[i].arg[v] = T_STR then
+          argstring := argstring + 'T_STR';
+        argstring := argstring + ', ';
+        inc(v);
+      end;
+
+      if argstring = '' then
+        argstring := argstring + 'T_NONE';
+      opcode := opcodelist[i].name;
+      break;
+    end;
+  end;
+  if (length(argstring) > 0) and (argstring <> 'T_NONE') then
+    SetLength(argstring, length(argstring)-2);
+  if argstring <> '' then
+    argstring := ' <' + opcode + '> ' + argstring;
+  TextEdit.Hint := 'Line: ' + inttostr(currentline) + argstring;
+end;
+
+procedure TfmScriptTE.TextEditMouseDown(Sender: TObject; Button: TMouseButton;
+  Shift: TShiftState; X, Y: Integer);
+begin
+  if Button = mbRight then
+    PopupMenu1.Popup(mouse.CursorPos.X, mouse.CursorPos.Y);
+end;
+
+procedure TfmScriptTE.Timer1Timer(Sender: TObject);
+var
+  lineposition: TTextEditorTextPosition;
+begin
+  lineposition.Line := 0;
+  if TextEdit.GetTextPositionOfMouse(lineposition) then
+    currentline := lineposition.Line;
+  TextEditClick(Timer1);
+end;
+
+procedure TfmScriptTE.Undo1Click(Sender: TObject);
+begin
+  TextEdit.DoUndo;
+end;
+
+procedure TfmScriptTE.Z100Click(Sender: TObject);
+var
+  Reg: TRegistry;
+begin
+  TextEdit.Zoom(100);
+  Z100.Checked := true;
+  Z125.Checked := false;
+  Z150.Checked := false;
+  Z200.Checked := false;
+  Z300.Checked := false;
+
+  Reg := TRegistry.Create;
```

Comment:
make this a function with the value to set as param so you can reuse it for z100, 125, 150 200 and 300